Added to reencrypt route example for default certs

bfallonf opened the pull request:
For: https://trello.com/c/MsyzMf0g/501-document-support-defaulting-the-destination-ca-certificate-to-the-service-serving-cert-address

cc: @smarterclayton 

Not sure if this is adequate. Will ask for ack over email.

smarterclayton commented on architecture/core_concepts/routes.adoc:
```diff
@@ -744,7 +744,7 @@ spec:
 <3> The `*destinationCACertificate*` field specifies a CA certificate to
 validate the endpoint certificate, securing the connection from the router to
-the destination. This field is required, but only for re-encryption.
+the destination. This field is required for re-encryption. However, it can be left empty to use a configurable default destination CA certificate path. For example, an existing router, logging, metrics, web console, or registry console certificate.
```
The "for example" probably doesn't make sense in this case.

Starting in 3.6, the router now automatically leverages the CA that is generated for service serving certificates (and is injected into every pod as `/var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt`) for routes that don't have a destinationCACertificate set.  This allows routes to be created that leverage end-to-end encryption without generating a certificate up front for the route.  Clusters where custom routers or the F5 router are in use may not support destinationCACertificates and will not admit the route unless the administrator has allowed that.

smarterclayton replied:
I would add a note to the service serving certs section.

bfallonf commented:
Thanks @smarterclayton . I've updated from your suggestions. I've also added the `DEFAULT_DESTINATION_CA_PATH` variable to the configuration parameters section. Please let me know if you have any more suggestions.

smarterclayton commented on architecture/core_concepts/routes.adoc:
```diff
@@ -742,9 +743,9 @@ spec:
-<3> The `*destinationCACertificate*` field specifies a CA certificate to
-validate the endpoint certificate, securing the connection from the router to
-the destination. This field is required, but only for re-encryption.
+<3> The `*destinationCACertificate*` field is required for re-encryption, and
```
The cert is only required if the admin didn't configure a default serving certificate.  Instead would say:

> ... is required for re-encryption, ..., securing the connection from the router to the destination pods.  If the service is using a service signing certificate, or the administrator has specified a default CA certificate for the router and the service has a certificate signed by that CA, this field may be omitted.

bfallonf commented:
Thanks @smarterclayton . I've pretty much changed as suggested. Please let me know if there's anything further.

Meanwhile, @openshift/team-documentation ready for peer review, please.

bmcelvee commented:
LGTM 🌟

bfallonf commented:
Thanks @bmcelvee ! I'll move this to QA.

bfallonf commented:
[rev_history]
|xref:../architecture/core_concepts/routes.adoc#architecture-core-concepts-routes[Core Concepts -> Routes]
|Added information about the `DEFAULT_DESTINATION_CA_PATH` environment variable.
%
